### DISCUSSION TITLE
Change refresh-picker value from within panel

### DISCUSSION THREAD

**Discussion by DavidMarom** [grafana/grafana#41471] [Dashboards, Panels & Visualizations]
I want to make a button inside my panel, that changes the refresh-picker to ‘5s’.
(My goal is to trigger the change also according to a certain logic and not only by a button click...)
![refresh-rate](https://user-images.githubusercontent.com/12570530/140923908-555a5855-5bae-4c12-abb5-c47968ac9b18.png)

Is there already a way of doing so?

**Comment by DavidMarom** [comment]
Got it:
import { getTimeSrv } from 'app/features/dashboard/services/TimeSrv';
getTimeSrv().setAutoRefresh('5s');

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
